CodeceptJS_PracticalTasks/practical_21_LoginTest/main.ts

```diff
@@ -0,0 +1,43 @@
+import { LoginPage } from "./LoginPage";
+import { ClientSidePage } from "./ClientSidePage";
+
+Feature("main");
+
+Scenario("PZ21 @C02101 LoginTest", async ({ I }) => {
+  const loginPage = new LoginPage(I);
+  const clientSidePage = new ClientSidePage();
+  //открыть страницу логина;
+  I.amOnPage("");
+  //проверить что видна форма для авторизации;
+  I.waitForVisible(loginPage.loginForm, 5); //лучше использовать для подвисаний (по умолчанию сек)
+  //проверить что в CS 1 подпункт, а в SS 6;
+  await loginPage.assertNumbersOfVisibleElements(loginPage.clientSideLoginContainer, 1);
+  await loginPage.assertNumbersOfVisibleElements(loginPage.serverSideLoginContainer, 6);
+  //проверить что все названия вкладок в CS и SS отображаются правильно (нет ошибок типа EvEnTStrEAm и т.д.);
+  //(проверить проход по масиву server side c проверкой по циклам по каждому элемента);
+  const serverSide = await I.grabTextFromAll(loginPage.serverSideLoginContainer);
```

Comment:
This is unnecessary to create variable only to transport in one place.
```
loginPage.checkElements(await I.grabTextFromAll(loginPage.serverSideLoginContainer), false);
loginPage.checkElements([await I.grabAttributeFrom(loginPage.clientSideLoginContainer, "title")]);
```

Reply:
Code was updated.